src/molecules/FileUpload/FileUpload.native.js

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from 'react';
+import { TouchableOpacity, LayoutAnimation } from 'react-native';
+import styled from 'styled-components/native';
+import DocumentPicker from 'react-native-document-picker';
+import PropTypes from 'prop-types';
+import View from '../../atoms/View';
+import Text from '../../atoms/Text';
+import Size from '../../atoms/Size';
+import Flex from '../../atoms/Flex';
+import Space from '../../atoms/Space';
+import Icon from '../../atoms/Icon';
+import ProgressBar from './ProgressBar';
+
+const MAX_PROGRESS_VALUE = 100;
+
+const DashedButton = styled(TouchableOpacity)`
+  background-color: ${(props) => props.theme.colors.background[200]};
+  border: 1px dashed ${(props) => props.theme.colors.tone[900]};
+  border-radius: 2px;
+`;
+
+const UploadContainer = styled(View)`
+  border: 1px solid ${(props) => props.theme.colors.tone[900]};
+  background-color: ${(props) => props.theme.colors.background[200]};
+  border-radius: 4px;
+`;
+
+const FileUpload = ({
+  title,
+  progress,
+  fileName,
+  fileTypes,
+  errorText,
+  helpText,
+  onFileSelectionError,
+  onFileSelected,
+  onFileRemoved,
+}) => {
+  const [file, setFileName] = useState(fileName);
+  const hasUploadCompleted = progress >= MAX_PROGRESS_VALUE;
+
+  const handleFileRemoval = () => {
+    // setFileName('');
+    onFileRemoved();
+  };
+
+  useEffect(() => {
+    if (fileName) {
+      setFileName(fileName);
+    } else if (!fileName) {
+      setFileName('');
+    }
+    LayoutAnimation.easeInEaseOut();
+  }, [fileName]);
+
+  const handleFilePick = async () => {
+    try {
+      const res = await DocumentPicker.pick({
+        type: !fileTypes
+          ? DocumentPicker.types.allFiles
+          : Object.keys(DocumentPicker.types).filter((type) => fileTypes.includes(type)),
```

Comment:
Why do we need to filter things out?

Reply:
filter allows only specific type of files to be selected. Like user wants to select only `pdf` and `doxx` from gallery, and not `.jpeg`, so jpeg files will be disabled.